Formula/vapor.rb

```diff
@@ -0,0 +1,22 @@
+class Vapor < Formula
+  desc "Command-line tool for Vapor (Server-side Swift web framework)"
+  homepage "https://vapor.codes"
+  head "https://github.com/vapor/toolbox.git"
+
+  depends_on :xcode => "11.4"
+
+  stable do
+    url "https://github.com/vapor/toolbox/archive/18.0.0.tar.gz"
+    sha256 "1d1b72d92dfb4e417dbbfe8486149f64a92cbfcf86db5209686fdd926b43c74c"
+  end
+
+  def install
+    system "swift", "build", "--disable-sandbox", "-c", "release", "-Xswiftc", "-cross-module-optimization"
+    mv ".build/release/vapor", "vapor"
+    bin.install "vapor"
+  end
+
+  test do
+    system "vapor", "-h"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.